src/arrayMethodJoin.js

```diff
@@ -6,6 +6,34 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    let joined = '';
+    let separ = ',';
+
+    if (!(separator === undefined)) {
+      separ = separator + '';
+    }
+
+    if (!(this[0] === null || this[0] === undefined)) {
+      joined = this[0] + '';
+    }
+
+    if (this.length === 1) {
+      return joined;
+    }
```

Comment:
remove these 3 if-blocks, you can do the same inside for-loop (just check for the last element)
if it's the last element in the array, you don't need to add separator after it (we can understand that the element is last if its index === this.length - 1)